tutorials/python/python-type-driven-development/tutorial.md

```diff
@@ -0,0 +1,209 @@
+
+In a [previous article][types-vs-tests], we discussed the benefits of working with _Type-Driven Development_ and the importance of static type checking in compilation time for strongly typed languages such as Java or Haskell. It doesn’t matter if the paradigm is object-oriented or functional, type-checking adds a layer of safety, reducing the amount of tests we need to write, and reduces our dependency on tests.
+
+## What about Python?
+
+Is it possible to use a strong type system in a language like Python? The answer is yes. [PEP-484][pep-484] introduced _Type Annotations to Python_ 3.5 back in 2014. It’s been a while, right? Why is this not popular? Because Python doesn’t require the code has types everywhere to run, they’re completely optional and mostly used by editors and IDEs. Another reason is that Python code is well-known to be readable in the least number of possible lines of code, but adding types to your code doesn’t make it more readable, what a paradox!
+
+In this article, we’ll be walking through the Python type system, how to add types to our code and Docstrings and perform static type checking using [mypy][mypy]. There are plenty options for static type-checking in Python, but in this tutorial we will follow [Guido van Rossum suggestion][guido-mypy].
+
+### Mypy
+
+Mypy is a third-party Python library that provides optional static type checking. Unlike other non-dynamic programming languages like Java, where the static type-checking takes place at compilation time, Mypy CLI does the type-check to a file (or a set of files), so it's up to you to decide where in your workflow the type-checking will be executed, it can be on CI or a pre-commit hook, just to mention some.
+
+To start using mypy, install it using any version of `pip` globally or in your virtual environment.
+
+```bash
+pip install mypy
+```
+
+To perform static type checking to a source file:
+
+```bash
+mypy my_file.py
+```
+
+## Primitive types
+
+Python is a multi-paradigm dynamic language, so the type of a variable is determined based on the value. This might be confusing, but it doesn't mean that Python has _weak typing_. Python is _strong typed_. Surprised? Check this out:
+
+```python
+movie = "Die Hard"
+movie = movie + 2
+# TypeError: cannot concatenate 'str' and 'int' objects
+```
+
+That said, let’s begin this tutorial with a basic example:
+
+```python
+meat = "Ground beef"
+print(type(meat))
+# <class 'str'>
+
+weight_pounds = 0.5
+print(type(weight_pounds))
+# <class 'float'>
+```
+
+In the above example, we are using the `type` function to inspect the type representation of any variable. What if we include types to our first example code? Let’s check the following example:
+
+```python
+meat: str = "Ground beef"
+weight_pounds: str = 0.5
+```
+
+If we execute this file we will expect the same output as the previous one, but when we run `mypy` we will get the following message:
```

Comment:
If we execute this file we expect the same output as the previous one, but we actually get the following: